include/alp/reference/vector.hpp

```diff
@@ -493,35 +494,35 @@ namespace alp {
 			/**
 			 * Constructor for a view over another functor-based vector.
 			 *
-			 * @tparam TargetType  The type of the target vector.
+			 * @tparam SourceType  The type of the target vector.
 			 *
 			 */
 			template<
-				typename TargetType,
+				typename SourceType,
 				std::enable_if_t<
-					std::is_same< TargetType, typename View::applied_to >::value &&
+					std::is_same< SourceType, typename View::applied_to >::value &&
 					internal::is_view_over_functor< View >::value &&
 					!internal::requires_allocation< View >::value
 				> * = nullptr
 			>
-			Vector( TargetType &target_vector, ImfR imf_r, ImfC imf_c ) :
+			Vector( SourceType &target_vector, ImfR imf_r, ImfC imf_c ) :
 				base_type( getFunctor( target_vector ), imf_r, imf_c ) {}
```

Comment:
Does Matrix have such constructor?

Reply:
I suggest to address this during the clean-up of views on functor-based containers.